src/libsyntax/parse/parser.rs

```diff
@@ -2323,6 +2324,16 @@ impl<'a> Parser<'a> {
                         self.recover_stmt();
                     }
                 }
+                if self.check(&token::Comma) {
+                    let mut err = self.sess.span_diagnostic.mut_span_err(
+                        exp_span.to(self.prev_span),
+                        "cannot use a comma after struct expansion",
+                    );
+                    err.span_suggestion_short(self.span, "remove this comma", "".to_owned());
+                    err.note("the struct expansion must always be the last field");
```

Comment:
"the" shouldn't be here if we talk about struct expansions in general?

Reply:
Actually, this is the first time I see the name "struct expansion".
Looks like it was a @Xion's invention in https://github.com/rust-lang/rust/issues/41834.

Reply:
I'm not even sure how this thing is called (`..expr` specifically, the whole `Struct { field1, field2, ..expr }` is usually called a "functional record update").
What the book/reference/etc say?

Reply:
The [book](https://doc.rust-lang.org/book/second-edition/ch05-01-defining-structs.html) refers to this as "field init shorthand", or "field init syntax", however still not sure how we call the item. The [first edition of the book](https://doc.rust-lang.org/book/first-edition/structs.html) refers to all this as "update syntax", so I'm not even sure if that is the official name.

Reply:
Ok, sounds good.